ios-base/Extensions/ViewExtension.swift

```diff
@@ -55,70 +55,76 @@ extension UIView {
   }
   
   // MARK: Constrains Helper
-  
+
+  func addSubviews(subviews: [UIView]) {
+    for subview in subviews {
+      addSubview(subview)
+    }
+  }
+
   func attachHorizontally(
-    subview: UIView,
+    to view: UIView,
     leadingMargin: CGFloat = UI.Defaults.margin,
     trailingMargin: CGFloat = UI.Defaults.margin
   ) {
     NSLayoutConstraint.activate([
-      subview.leadingAnchor.constraint(equalTo: leadingAnchor, constant: leadingMargin),
-      subview.trailingAnchor.constraint(
-        equalTo: trailingAnchor,
+      leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: leadingMargin),
+      trailingAnchor.constraint(
+        equalTo: view.trailingAnchor,
         constant: -trailingMargin
       )
     ])
   }
-  
-  func addSubviews(subviews: [UIView]) {
-    for subview in subviews {
-      addSubview(subview)
-    }
-  }
-  
+
   func attachVertically(
-    subview: UIView,
+    to view: UIView,
     topMargin: CGFloat = UI.Defaults.margin,
     bottomMargin: CGFloat = UI.Defaults.margin
   ) {
     NSLayoutConstraint.activate([
-      subview.topAnchor.constraint(equalTo: topAnchor, constant: topMargin),
-      subview.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -bottomMargin)
+      topAnchor.constraint(equalTo: view.topAnchor, constant: topMargin),
+      bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -bottomMargin)
     ])
   }
   
-  /// Center a subview already added to its parent view
+  /// Centers the view horizontally  in a specific view
   ///
   /// - Parameters:
-  ///   - subview: UIView to be centered on its parent
-  ///   - verticalOffset: CGFloat indicating the vertical displacement of the view
-  ///   - horizontalOffset: CGFloat indicating the horizontal displacement of the view
-  ///   - vertically: Boolean that indicates if the view will be center vertically
-  ///   - horizontally: Boolean that indicates if the view will be center horizontally
-  func center(
-    subview: UIView,
-    verticalOffset: CGFloat = 0,
-    horizontalOffset: CGFloat = 0,
-    vertically: Bool = true,
-    horizontally: Bool = true
+  ///   - view: UIView on which the view will be centered horizontally
+  ///   - withOffset: CGFloat indicating the horizontal displacement of the view
+  func centerHorizontallyIn(
+    _ view: UIView,
+    withOffset offset: CGFloat = 0
   ) {
-    if !subviews.contains(subview) {
-      addSubview(subview)
-    }
-    
-    if vertically {
-      subview.centerYAnchor.constraint(
-        equalTo: centerYAnchor,
-        constant: verticalOffset
-      ).isActive = true
+    guard superview == view else {
```

Comment:
this isn't needed IMO. you may want to center sibling views

Reply:
please add

```
func center(_ view: UIView, withOffset offset: CGPoint = .zero) {
   centerHorizontallyIn(view, withOffset: offset.x)
   centerVerticallyIn(view, withOffset: offset.y)
}

Reply:
> this isn't needed IMO. you may want to center sibling views

We need this to be sure that the view is already added to that view. We could change this to check if the view is a subview of the view that's coming through param otherwise this will crash.

Another option is to chang the method to be `centerHorizontallyInSuperview` then no have `view` parameter and just check inside if `superview` is not nil

Reply:
you may have View A and B, both subviews of View C. And you may want to center view A with View B.
This would be possible without the current check. I think its responsibility of the client to see that both View A and B are in the same view hierarchy.